File: PYTHON/lector.py
from time import time
import Algoritmos

#w = open("BubblePy.txt","w")
#w = open("CountingPy.txt","w")
#w = open("HeapPy.txt","w")
#w = open("InsertionPy.txt","w")
#w = open("MergePy.txt","w")
#w = open("QuickPy.txt","w")
w = open("SelectionPy.txt","w")

NUM_PRUEBAS=int(input())
w.write(str(NUM_PRUEBAS)+"\n")

numpuntos=15

for ii in range(numpuntos):
	DIM_ARREGLO=int(input())
	tiempo=str(DIM_ARREGLO)
	for i in range(NUM_PRUEBAS):
		aux=input()
		aux=aux.split()
		arreglo=[]
		for j in arreglo:
			arreglo·append(int(j))

		start_time = time()
		#Algoritmos.BubbleSort(arreglo)
		#Algoritmos.countSort(arreglo)
		#Algoritmos.HeapSort(arreglo)
		#Algoritmos.InsertionSort(arreglo)
		#Algoritmos.MergeSort(arreglo)
		#Algoritmos.QuickSort(arreglo)
		Algoritmos.SelectionSort(arreglo)

		elapsed_time = time()-start_time
		tiempo=tiempo+" "+str( elapsed_time*1000000000 );
	w.write(tiempo+"\n")

w.close()

File: GOLANG/lector.go
package main
import "fmt"
import "time"
import "strconv"
import "os"
import "./Algoritmos"

func main() {
   var NUM_PRUEBAS int
   fmt.Scan(&NUM_PRUEBAS)
   var numpuntos int
   numpuntos=15
   var DIM_ARREGLO int
   var aux int
   var tiempo string
   f, err := os.Create("CountingGo.txt")
   //f, err := os.Create("HeapGo.txt")
   //f, err := os.Create("InsertionGo.txt")
   //f, err := os.Create("QuickGo.txt")

   if err != nil {
       fmt.Println(err)
       return
   }
   f.WriteString(strconv.Itoa(NUM_PRUEBAS)+"\n")
   for ii := 0; ii < numpuntos; ii++ {
      fmt.Scan(&DIM_ARREGLO)
      tiempo=""+strconv.Itoa(DIM_ARREGLO)
      for i:=0;i<NUM_PRUEBAS;i++{
         var array []int
         for j:=0;j<DIM_ARREGLO;j++{
            fmt.Scan(&aux)
            array=append(array,aux)    
         }
         start := time.Now()
         array=Algoritmos.CountingSort(array)
         //array=Algoritmos.HeapSort(array)
         //array=Algoritmos.InsertionSort(array)
         //array=Algoritmos.QuickSort(array,0,len(array)-1)
         elapsed := time.Since(start)
         tiempo=tiempo+" "+strconv.Itoa(int(elapsed));
      }
      tiempo=tiempo+"\n"
      //fmt.Println(tiempo)
      f.WriteString(tiempo)
   }

}



